course3/homework-string/my_string.cc

```diff
@@ -0,0 +1,101 @@
+#include "my_string.h"
+
+#include <cstring>
+#include <stdexcept>
+
+MyString::MyString(): mSize(0) { }
+
+MyString::MyString(const char *inchar) :mSize(std::strlen(inchar)), mBuffer(new char[mSize + 1])
+{
+	std::strcpy(mBuffer.get(), inchar);
+}
+
+MyString::MyString(const std::size_t size) :mSize(size), mBuffer(new char[mSize + 1])
```

Comment:
This is unusual.
1. What is the content of the string after this constructor is used?
2. It allows for the following code:
```
MyString sss = 5;
// what is in sss now???
```
and also for this:
```
void someFunction(const MyString& s);

someFunction(42);
```